### ISSUE TITLE
css module @value breaks when prettier puts each item on a new line

### ISSUE THREAD

**Issue by zyzski** [prettier/prettier#2355]
**works:** 
`@value fontStack: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;`

**breaks:**
![screen shot 2017-06-29 at 11 56 37 am](https://user-images.githubusercontent.com/4314335/27697416-326b36b0-5cc2-11e7-8096-fcc2c79c4bf3.png)

**Comment by vjeux** [comment]
Hey, could you provide an example on the playground on what you expect vs what happened? https://prettier.io/

Thanks!

**Comment by zyzski** [comment]
Sorry, meant to add more detail in an edit but forgot to post. 

Expected behavior would be leaving it on a single line, because the value doesn't parse when it's multiline.

I suppose this bug could also be filed with the CSS modules project to allow multi line comma separated variable values. 

Feel free to close if you think it should be their responsibility.

Temp fix would be to toggle prettier when working with CSS module variables in react.

**Comment by vjeux** [comment]
You can add `/* prettier-ignore */` on the line before to get you unblocked for now.

**Comment by vjeux** [comment]
Can you raise an issue on the css-modules project? We're already adding hacks in prettier to get some of the things working, but I'd like to avoid having to add them all over the place if possible. https://github.com/prettier/prettier/pull/2190

**Comment by zyzski** [comment]
will do, thank you. appreciate the hard work you do for this project.

**Comment by zaneriley** [comment]
CSS modules supports multiline `@values` now, but you need a wrapping parens:

```css
/* working multiline support syntax */
@value (value1,
  value 2,
  value 3,
  value 4,
  value 5) from "some-other-file.css";
```

However, Prettier doesn't seem to do this out of the box, so:
```css
/* Pre-formatted code */
@value value1, value 2, value 3, value 4, value 5 from "some-other-file.css";
```
becomes:

```css
/* Formatted code from Prettier */
@value value1,
  value 2,
  value 3,
  value 4,
  value 5 from "some-other-file.css";
```

which doesn't work. 

@vjeux Could we get Prettier to add the parens to `@values` if they get formatted to multiline? The original problem (multiline support) got added to CSS modules, but since Prettier is handling formatting it seems like it the parens should be handled here.

Thanks for all the hard work!

**Comment by azz** [comment]
@zaneriley I just had an attempt at making this change. Due to the CSS AST being super generic it is difficult to get that formatting. Closest I got was 5376851.

```css
@value (
  value1,
  value 2,
  value 3,
  value 4,
  value 5,
  value 6,
  value 7,
  value 8 from "some-other-file.css"
);
```

Moving the `)` before the `from` is proving difficult.

**Comment by LukeChannings** [comment]
Does the suggested fix work for this example?

```css
@value font-bold: "Helvetica Neue", Helvetica, Arial, sans-serif;
```

I am using this in my code for font family definitions (not importing), and it's being reformatted and breaks.

**Comment by James-E-Adams** [comment]
Any news on this? Is wrapping the imported values in parens the only way to resolve this atm?

**Comment by j-f1** [comment]
You can also put `/* prettier-ignore */` on the line before the declaration to make Prettier print it as-is.

**Comment by vjeux** [comment]
cc @evilebottnawi

**Comment by evilebottnawi** [comment]
@vjeux it is not easy, we should white parser over @value, i can do this in this repo, but it is sound ugly

**Comment by kumarharsh** [comment]
Is there any movement on this? Which project would have the responsibility for implementing this feature? Sadly, I've had to disable prettier on css due to this issue :-(

**Comment by evilebottnawi** [comment]
@kumarharsh PR welcome

**Comment by kumarharsh** [comment]
I am willing to try, but I want to know whether I should try adding this functionality in prettier or some other project. Because @azz's attempt above highlights an issue with prettier not having enough information about the CSS AST.

**Comment by evilebottnawi** [comment]
@kumarharsh we can implement function what return value tokens between `@value` and `from`

**Comment by robert-j-webb** [comment]
I'm not seeing this anymore: [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEABAbgQwDYFc4AEWecAjADRE74EBMlxNAzA9YQCyskECsXNANn6EA7MIIAOAgDMAThAC2BADogAzorgBaCDAAWcWVukBLbHAB0YNWtUBuZVBDkQEAA4wT0NclCZZ8gDuAAr+CD4omOgQJgAmziAARrKYYADWcDAAym6pJlAA5sgwsvguejAK2ADqeibwarlgcFnh9Sbo9QCeyOA2CflqhjDBKQUKmMjSOEMuAFZqAB4AQinpmVmYCnAAMvlwUzNw80tZ+QXmAIq4ugdI09izILmyQ7K9atY+Lm6y+TDVOL6ZASAAMP3kQ2qKTcvV+cDe6AOLgAjjd4KN3BEQJg1FooHA4LEiQlZHA0SYyaNMONJvcji4hgoTMVSsd1Ocrui7g8njBMIlAbFgUh6CASpgzOcAMKKCa9KDQZEgXBDAAqAoivPZ6HwAEkoMTYFkwH8PABBQ1ZGBdcyHR5wAC+jqAA)

It still doesn't add new lines, but it's not mangling.

```scss
@value value1, value 2, value 3, value 4, value 5, value 6, value 7, value 8 from "some-other-file.css";
```

Becomes
```scss
@value value1, value 2, value 3, value 4, value 5, value 6, value 7, value 8 from "some-other-file.css";
```

**Comment by j-f1** [comment]
Closing. If anyone sees this type of behavior, please open a new issue.
